File: exercise/kotlin/day19/src/main/kotlin/travel/SantaTravelCalculator.kt
package travel

object SantaTravelCalculator {
    fun calculateTotalDistanceRecursively(numberOfReindeers: Int): Int {
        if (numberOfReindeers == 1) return 1
        (2 * calculateTotalDistanceRecursively(numberOfReindeers - 1) + 1)
            .let { result ->
                if (result == -1) throw ArithmeticException("Integer overflow")
                return result
            }
    }
}

File: exercise/scala/day19/src/main/scala/travel/SantaTravelCalculator.scala
package travel

object SantaTravelCalculator {
  def calculateTotalDistanceRecursively(numberOfReindeers: Int): Int = {
    if (numberOfReindeers == 1) 1
    else {
      val result = 2 * calculateTotalDistanceRecursively(numberOfReindeers - 1) + 1
      if (result < 0) throw new ArithmeticException("Integer overflow")
      result
    }
  }
}